crypto/src/bulletproofs.rs

```diff
@@ -0,0 +1,263 @@
+use std::ops;
+
+use bulletproofs::{BulletproofGens, PedersenGens, RangeProof};
+use curve25519_dalek_ng::{
+    ristretto::{CompressedRistretto, RistrettoPoint},
+    scalar::Scalar,
+};
+use merlin::Transcript;
+use once_cell::sync::OnceCell;
+use serde::{de, Deserialize, Serialize};
+
+use crate::traits::ToFromBytes;
+
+//
+// Pedersen commitments
+//
+
+#[derive(Debug, Clone)]
+pub struct PedersenCommitment {
+    point: RistrettoPoint,
+    bytes: OnceCell<[u8; 32]>,
+}
+
+impl PedersenCommitment {
+    ///
+    /// Creates a new Pedersen commitment from a value, and a blinding factor
+    ///
+    pub fn new(value: [u8; 32], blinding_factor: [u8; 32]) -> Self {
+        let generators = PedersenGens::default();
+        let value = Scalar::from_bits(value);
+        let blinding = Scalar::from_bits(blinding_factor);
+        let point = generators.commit(value, blinding);
+
+        PedersenCommitment {
+            point,
+            bytes: OnceCell::new(),
+        }
+    }
+}
+
+impl ops::Add<PedersenCommitment> for PedersenCommitment {
+    type Output = PedersenCommitment;
+
+    fn add(self, rhs: PedersenCommitment) -> PedersenCommitment {
+        PedersenCommitment {
+            point: self.point + rhs.point,
+            bytes: OnceCell::new(),
+        }
+    }
+}
+
+impl ops::Sub<PedersenCommitment> for PedersenCommitment {
+    type Output = PedersenCommitment;
+
+    fn sub(self, rhs: PedersenCommitment) -> PedersenCommitment {
+        PedersenCommitment {
+            point: self.point - rhs.point,
+            bytes: OnceCell::new(),
+        }
+    }
+}
+
+impl AsRef<[u8]> for PedersenCommitment {
+    fn as_ref(&self) -> &[u8] {
+        self.bytes.get_or_init(|| self.point.compress().to_bytes())
+    }
+}
+
+impl ToFromBytes for PedersenCommitment {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, signature::Error> {
+        if bytes.len() != 32 {
```

Comment:
Could you please extract the number `32` to a const so we avoid [magic numbers](https://www.codebyamir.com/blog/software-anti-patterns-magic-numbers#:~:text=A%20magic%20number%20is%20a,so%20they%20should%20be%20avoided.) ?

Reply:
Yes let's encode this as const ped commitment length